oauth.go

```diff
@@ -0,0 +1,82 @@
+package twilio
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/golang-jwt/jwt"
+	"github.com/twilio/twilio-go/client"
+	iam "github.com/twilio/twilio-go/rest/iam/v1"
+)
+
+type TokenAuth struct {
+	token string
+	OAuth client.OAuth
+}
+
+func (t *TokenAuth) NewTokenAuth(token string, o client.OAuth) *TokenAuth {
+	return &TokenAuth{token: token, OAuth: o}
+}
+
+func (t *TokenAuth) FetchToken() (string, error) {
+	if t.token != "" && !t.TokenExpired() {
+		return t.token, nil
+	}
+
+	token, err := t.OAuth.GetAccessToken(context.Background())
+	if err != nil {
+		return "", err
+	}
+
+	t.token = token
+	return t.token, nil
+}
+
+func (t *TokenAuth) TokenExpired() bool {
```

Comment:
Given you already have a Token, naming this `TokenExpired` is redundant - can just name it `Expired()` 

```suggestion
func (t *TokenAuth) Expired() bool {
```